substrate/client/consensus/babe/src/authorship.rs

```diff
@@ -108,7 +108,8 @@ pub(super) fn secondary_slot_author(
 		return None
 	}
 
-	let rand = U256::from((randomness, slot).using_encoded(sp_crypto_hashing::blake2_256));
+	let rand =
+		U256::from_little_endian(&(randomness, slot).using_encoded(sp_crypto_hashing::blake2_256));
```

Comment:
wasn't `From::from` based on `from_big_endian`?
see https://github.com/paritytech/parity-common/pull/859/files
this might create consensus problems with some nodes upgrading to this code

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5673/files#diff-01d2862c9d91433a2f2d36fbd35ac059786ce6daa7b09557cb0203f4a5031f8d

cc @pgherveou

Reply:
I think this is causing the failures in some tests.

```
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540617
https://grafana.teleport.parity.io/goto/P4omWXkHR?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540639
https://grafana.teleport.parity.io/goto/8xYZZXkNg?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540550
https://grafana.teleport.parity.io/goto/WvoIZXkNR?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540540
https://grafana.teleport.parity.io/goto/0uUHWuzHR?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540189
https://grafana.teleport.parity.io/goto/6PeOZXkHR?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7540084
https://grafana.teleport.parity.io/goto/jWV5ZXzNR?orgId=1

--
https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/7539899
https://grafana.teleport.parity.io/goto/yjEpWukHR?orgId=1
```

Reply:
https://github.com/paritytech/polkadot-sdk/pull/5994